errutil.go

```diff
@@ -108,6 +108,27 @@ func New(t Tags) error {
 	return NewFrameError(Caller(1), t, nil, false)
 }
 
+// WithStack returns a new error with complete stack information from
+// the caller of WithStack.
+//
+// This is useful in cases where errors aren't or can't be created and
+// passed around, such as when recovering from panics.
+func WithStack(err error) error {
+	frames := callers(1)
+	if len(frames) <= 1 {
+		return NewFrameError(frames[0], nil, err, false)
+	}
+	for i := 0; i < len(frames); i++ {
```

Comment:
nit, could use range

Reply:
Needs a go.mod bump to 1.22, can do that in a followup.

Reply:
I don't mean for `i := range num` but only `for _, f := range frames`

Reply:
Of course, yes. Think I've flipped the ordering so many times I was convinced the full form was necessary. 😅